packages/core/src/workflows/workflow.ts

```diff
@@ -330,6 +331,17 @@ export class Workflow<
 
       const suspendedPaths: Set<string> = new Set();
       this.#actor.subscribe(async state => {
+        if (this.#onStepTransition) {
+          this.#onStepTransition.forEach(onTransition => {
+            onTransition({
+              runId: this.#runId,
+              value: state.value as Record<string, string>,
+              context: state.context as WorkflowContext,
+              activePaths: this.#getActivePathsAndStatus(state.value as Record<string, string>),
+              timestamp: Date.now(),
+            });
+          });
```

Comment:
if it becomes a Set we can change this to
```suggestion
          for (const onTransiition of this.#onStepTransition) {
            onTransition({
              runId: this.#runId,
              value: state.value as Record<string, string>,
              context: state.context as WorkflowContext,
              activePaths: this.#getActivePathsAndStatus(state.value as Record<string, string>),
              timestamp: Date.now(),
            });
          };
```

Reply:
this still works with `.forEach()`